libopendroneid/wifi.c

```diff
@@ -0,0 +1,340 @@
+/*
+Copyright (C) 2019 Intel Corporation
+
+SPDX-License-Identifier: Apache-2.0
+
+Open Drone ID C Library
+
+Maintainer:
+Gabriel Cox
+gabriel.c.cox@intel.com
+*/
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <errno.h>
+#include <byteswap.h>
+
+#include "opendroneid.h"
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define cpu_to_le16(x)  (x)
+#else
+#define cpu_to_le16(x)      (bswap_16(x))
+#endif
+
+#define IEEE80211_FCTL_FTYPE          0x000c
+#define IEEE80211_FCTL_STYPE          0x00f0
+
+#define IEEE80211_FTYPE_MGMT            0x0000
+#define IEEE80211_STYPE_ACTION          0x00D0
+
+
+char *drone_export_gps_data(ODID_UAS_Data *UAS_Data)
+{
+	int len = 0, total_len = 8192;
+	char *drone_str;
+
+	drone_str = malloc(total_len);
```

Comment:
Would it be possible to not use malloc but instead allocate the memory somewhere higher above and pass it in?
On Linux systems this would be fine but I guess this code could be used in deeply embedded systems/drones as well where malloc isn't necessarily available.